python/spacewalk/satellite_tools/reposync.py

```diff
@@ -1152,6 +1152,9 @@ def import_packages(self, plug, source_id, url, is_non_local_repo):
         downloader.set_log_obj(logger)
         downloader.run()
 
+        log(0, 'Filtering packages that failed to download')
+        to_process = [i for i in to_process if i[0].getNVREA() not in downloader.failed_pkgs]
```

Comment:
Worst case this is `O(n^2)`, right? `downloader.failed_pkgs` will in most cases be a short list so this does not matter much, but a set could also make sense for `downloader.failed_pkgs`

Reply:
Makes sense; it was my assumption that the list of failed pkgs won't be long, but you're totally right about the set.